spec.html

```diff
@@ -4409,7 +4405,29 @@ <h1>Runtime Semantics: MV</h1>
               The MV of <emu-grammar>StrUnsignedDecimalLiteral ::: DecimalDigits ExponentPart</emu-grammar> is the MV of |DecimalDigits| times 10<sup>_e_</sup>, where _e_ is the MV of |ExponentPart|.
             </li>
           </ul>
-          <p>Once the exact MV for a String numeric literal has been determined, it is then rounded to a value of the Number type. If the MV is 0, then the rounded value is *+0*<sub>𝔽</sub> unless the first non white space code point in the String numeric literal is `-`, in which case the rounded value is *-0*<sub>𝔽</sub>. Otherwise, the rounded value must be the Number value for the MV (in the sense defined in <emu-xref href="#sec-ecmascript-language-types-number-type"></emu-xref>), unless the literal includes a |StrUnsignedDecimalLiteral| and the literal has more than 20 significant digits, in which case the Number value may be either the Number value for the MV of a literal produced by replacing each significant digit after the 20th with a 0 digit or the Number value for the MV of a literal produced by replacing each significant digit after the 20th with a 0 digit and then incrementing the literal at the 20th digit position. A digit is significant if it is not part of an |ExponentPart| and</p>
+          <p>For the MV of any production not shown above, see <emu-xref href="#sec-static-semantics-mv"></emu-xref>.</p>
+        </emu-clause>
+
+        <emu-clause id="sec-stringtonumber" aoid="StringToNumber">
+          <h1>StringToNumber ( _str_ )</h1>
+          <p>The abstract operation StringToNumber takes argument _str_ (a String) and returns a Number. It performs the following steps when called:</p>
+          <emu-alg>
+            1. Let _text_ be ! StringToCodePoints(_str_).
+            1. Let _literal_ be ParseText(_text_, |StringNumericLiteral|).
+            1. If _literal_ is a List of errors, return *NaN*.
+            1. If _literal_ contains a |StrUnsignedDecimalLiteral| and _literal_ has more than 20 significant digits, then
+              1. Let _lit_ be an implementation-defined choice of:
+                * _literal_
```

Comment:
By my reading, this option is not actually available to implementations in the previous spec text:

> the Number value may be either the Number value for the MV of a literal produced by replacing each significant digit after the 20th with a 0 digit or the Number value for the MV of a literal produced by replacing each significant digit after the 20th with a 0 digit and then incrementing the literal at the 20th digit position.

lists only two options, neither of which is using the MV of the actual literal.

That said, I'm almost certain that the MV of `_literal_` is necessarily equal to the MV of one of the other two options in this list, given the constraints of this section. So it doesn't really matter, but by the same token is probably confusing.

Reply:
I'd say the prose is ambiguous. E.g., consider:
- "Breakfast is cereal, unless it's Saturday, in which case breakfast may be pancakes."

Here, I think it's fairly clear that Saturday breakfast could still be cereal. (Otherwise, you'd say "... *is* pancakes".)

- "Breakfast is cereal, unless it's Saturday, in which case breakfast may be either pancakes or waffles."

Here, it's unclear if cereal is still an option for Saturday breakfast. It's possible to read the "may be" as in the previous sentence (so cereal is an option), and yet it's also possible to read it as 'referring' only to the choice  between pancakes and waffles (so cereal isn't an option).

Roughly, it's the difference between "might be" and "must be". In the spec text, it's unclear which is meant. (Note that "must be" is used earlier in the sentence, which suggests that it *would* have been used later if that *were* the appropriate sense, and hence that it's not the appropriate sense, "might be" is. But that's not very convincing.) The wording goes all the way back to the first edition.

> That said, I'm almost certain that the MV of _literal_ is necessarily equal to the MV of one of the other two options in this list, given the constraints of this section.

Yeah, me too, almost.

Reply:
Hm, I take @jmdyck's reading, that the MV of _literal_ is currently allowed for literals with more than 20 significant digits.